### DISCUSSION TITLE
DAG trigger wait_for_completion not working as expected?

### DISCUSSION THREAD

**Discussion by fiona-streetgroup** [apache/airflow#16583] [Q&A] [answered]
I have multiple DAGs that depend on `initial_dag` running first, after that I'd like the dependent DAGs to run one by one after each other. Here is an example what I have: 

    dag = DAG(
        dag_id=DAG_NAME,
        default_args=default_args,
        schedule_interval=None,
        start_date=airflow.utils.dates.days_ago(1)
    )

    initial_dag = BashOperator(
        task_id='initial_dag',
        bash_command="python /home/airflow/gcs/dags/task.py",
        dag=dag
    )

    dependent_dag1 = TriggerDagRunOperator(
        task_id="dependent_dag1",
        trigger_dag_id="dependent_dag1",
        wait_for_completion=True,
        dag=dag
    )

    dependent_dag2 = TriggerDagRunOperator(
        task_id="dependent_dag2",
        trigger_dag_id="dependent_dag2",
        wait_for_completion=True,
        dag=dag
    )

    dependent_dag3 = TriggerDagRunOperator(
        task_id="dependent_dag3",
        trigger_dag_id="dependent_dag3",
        wait_for_completion=True,
        dag=dag
    )

    initial_dag >> dependent_dag1 >> dependent_dag2 >> dependent_dag3

I thought the `wait_for_completion=True` would complete the run of each DAG before triggering the next one. E.g. `initial_dag` runs and completes, then trigger `dependent_dag1` and wait for that to complete to trigger subsequent tasks. 

The order the DAGs are being triggered is correct, but it doesn't seem to be waiting for the previous DAG to complete first, e.g. `dependent_dag2` is being triggered before `dependent_dag1` has been completed. 

Am I missing something here? 

We are running Airflow on GCP Cloud Composer with the following config:

* Image version: `composer-1.16.1-airflow-1.10.15`

**Comment by josh-fell** [comment, marked as answer]
Unfortunately, the `wait_for_completion` parameter isn't available as part of 1.10.x versions (see [documentation](https://github.com/apache/airflow/blob/v1-10-stable/airflow/operators/dagrun_operator.py)) and is being handled and ignored as a generic `kwarg`.  This parameter is available starting with 2.0.
